# capstone_DEanalysis
Differential expression TCGA dataset

This repository is a work in progress. The goal is to compare different DE analysis techniques e.g. manual (using Python to code the statistical tests) vs package based solutions (e.g. Bioconductor packages such as DESeq2). This is intended to begin a series of submissions comparing different approaches to solving common problems in bioinformatics. Following this analysis, we will look at single cell RNA-seq analysis (an increasingly popular laboratory technique using assays such as droplet based microfluidics), classification problems using tree based methods, and at least one implimentation of neural networks in the context of computational biology (using Tensorflow). All work will be done using Jupyter Notebooks (available in the Anaconda3 Navigator) or the Rstudio IDE. 

Bioinformatics is increasingly shifting to Python as the language of choice. While R has many more packages and a much longer lineage of providing solutions for bioinformatics, in many cases Python can increase efficiency and flexibility. Here, we hope to provide basic guidelines for many common tasks in bioinformatics, and demonstrate how they should be coded (in bot R and Python). We  will also discuss the strengths/weaknesses of each approach and quality control/testing.
